lib/puppet/provider/shellvar/augeas.rb

```diff
@@ -53,6 +53,19 @@ def unset_purge(aug)
     aug.rm("$target/@unset[count(*)=0]") if unset_seq?
   end
 
+  def exists?
+    augopen do |aug|
+      if resource[:ensure] == :absent and resource[:array_append] and not aug.match('$resource').empty?
+        is_str = value.is_a?(Array) ? value.join(' ') : value
+        is_arr = is_str.split(' ')
+        should_arr = Array(resource[:value])
+        not (is_arr - (is_arr - should_arr)).empty?
+      else
+        super
+      end
+    end
+  end
```

Comment:
I think I would rather patch `ensure`'s `insync?` to add (line 39 in the type):

```ruby
return true if should == :absent and provider.resource[:array_append] and provider.exists?
```

as this will avoid overriding `exists?`

Reply:
Right, that's easier! Changed it.